tutorials/03_better_tool/tut_better_tool/main.rb

```diff
@@ -0,0 +1,203 @@
+# Copyright 2021 Trimble Inc
+# Licensed under the MIT license
+
+require 'sketchup.rb'
+
+module Tutorials
+  module BetterTool
+
+    class LineTool
+
+      # Threshold in logical screen pixels for when the mouse is considered to
+      # be dragged.
+      APERTURE = 10
+
+      def activate
+        @mouse_ip = Sketchup::InputPoint.new
+        @picked_first_ip = Sketchup::InputPoint.new
+
+        # Track where mouse was pressed down so we can compare its position when
+        # later released.
+        @x_down = @y_down = nil
+
+        update_ui
+      end
+
+      def deactivate(view)
+        view.invalidate
+      end
+
+      # Enable editing in the measurement bar when the first point has been
+      # picked.
+      def enableVCB?
+        picked_first_point?
+      end
+
+      def resume(view)
+        update_ui
+        view.invalidate
+      end
+
+      def onCancel(reason, view)
+        reset_tool
+        view.invalidate
+      end
+
+      def onMouseMove(flags, x, y, view)
+        if picked_first_point?
+          @mouse_ip.pick(view, x, y, @picked_first_ip)
+
+          # Print the length of the previewed line to the measurement bar.
+          # Printing a Length object (as opposed to a float) automatically
+          # formats it according to the model units.
+          Sketchup.vcb_value = @mouse_ip.position.distance(@picked_first_ip.position)
+        else
+          @mouse_ip.pick(view, x, y)
+        end
+        view.tooltip = @mouse_ip.tooltip if @mouse_ip.valid?
+        view.invalidate
+      end
+
+      def onLButtonDown(flags, x, y, view)
+        # Track where in screen space mouse is pressed down.
+        @x_down = x
+        @y_down = y
+
+        if picked_first_point? && create_edge > 0
+          # When the user have picked a start point and then picks another point
+          # we create an edge and try to create new faces from that edge.
+          # Like the native tool we reset the tool if it created new faces.
+          reset_tool
+        else
+          # If no face was created we let the user chain new edges to the last
+          # input point.
+          @picked_first_ip.copy!(@mouse_ip)
+        end
+
+        update_ui
+        view.invalidate
+      end
+
+      def onLButtonUp(flags, x, y, view)
+        if (x - @x_down).abs > APERTURE || (y - @y_down).abs > APERTURE
+          # Mouse is released far enough away from where it was pressed to be
+          # considered to be dragged there.
+
+          create_edge
+
+          # When drag-drawing, it feels odd to chain line drawing and we
```

Comment:
I'm not clear on what you mean by "chain line drawing". Can you elaborate or rephrase?